### ISSUE TITLE
Add weight window parameters to cloudmc

### ISSUE THREAD

**Issue by moellep** [radiasoft/sirepo#6033]
Add support for weight windows. Choose a good default based on the input geometry bounds.

https://docs.openmc.org/en/stable/pythonapi/generated/openmc.WeightWindows.html

**Comment by sjcrs** [comment]
The weight windows PR was accepted into the OpenMC `develop` branch, but has not yet made it to a tagged release.  We'll have to wait until the build on jupyter picks it up. 

> [vagrant@rs30 ~]$ which openmc
~/.local/bin/openmc

>[vagrant@rs30 ~]$ openmc --version
OpenMC version 0.13.3

 Once the OpenMC changes propagate to jupyter, I will create a clean notebook example to guide the interface development.

**Comment by moellep** [comment]
This issue is a priority

**Comment by moellep** [comment]
Some good examples here:
https://github.com/fusion-energy/neutronics-workshop/blob/main/tasks/task_14_variance_reduction/2_shielded_room_single_ww.ipynb

https://github.com/fusion-energy/neutronics-workshop/blob/main/tasks/task_14_variance_reduction/3_sphere_iterative_per_run_ww.ipynb

**Comment by moellep** [comment]
I added a new Variance Reduction tab, and 3 different techniques. Below is a sim run with similar runtime and the results. Weight windows do a better job of filling out the geometry. There will need to be a few design iterations to get this featue right. One question does the "Normalize to particle count" calculation need to change when running with weight windows?

"None" no variance reduction
![none](https://github.com/radiasoft/sirepo/assets/6516910/123578b5-716a-4041-8aee-842799ab9299)
![Tally-Results-t2-absorption-mean (5)](https://github.com/radiasoft/sirepo/assets/6516910/0b3a625b-6d7c-479f-9079-3240ac4cb557)

"Survival Biasing"
![survival](https://github.com/radiasoft/sirepo/assets/6516910/1b9ea4b0-893c-4a30-a20d-7197295c27b0)
![Tally-Results-t2-absorption-mean (6)](https://github.com/radiasoft/sirepo/assets/6516910/4d22bd76-4fa9-4fe6-b9ba-44d07259e82c)

"Weight Windows for Tally" Select a tally and the number of iterations/particles to run each iteration. Runs with ```openmc.lib.run_in_memory```.
![wwtally](https://github.com/radiasoft/sirepo/assets/6516910/e5d24295-6b70-4be2-8b67-1c16f7d83719)
![Tally-Results-t2-absorption-mean (7)](https://github.com/radiasoft/sirepo/assets/6516910/94bbfd4a-893a-47a4-9e77-49eea28f823c)

"Weight Windows" manually define the mesh to be used for weight windows. For now it runs as a single simulation, rather than an iterative approach like "Weight Windows for Tally". It could be improved by creating a default mesh which covers all the tallies.
![wwgrid](https://github.com/radiasoft/sirepo/assets/6516910/adb0f2e0-644d-4439-a130-22424cd6c3d8)

**Comment by sjcrs** [comment]
Awesome! Can you please make a couple of changes:
 - Rename Weight Windows to "MAGIC Method", to distinguish it from the RandomRay/FW-CADIS method that John Tramm is developing
 - Test this with a tally that includes both a mesh filter and an energy filter

**Comment by sjcrs** [comment]
To the question of "normalize to particle count", I don't think it matters for the initial iterations, however those iterations should be running with fewer primary particles as the method converges on the proper weights.  We'd then expect the user to take those weight windows to a 'production' run, where the normalization should be applied.

**Comment by moellep** [comment]
I've updated the Variance Reduction methods to show "MAGIC Method" and "MAGIC Method for Tally".

I ran a simulation with an energy filter with 4 buckets and compared with a non-magic-method sim which ran for a similar amount of time. The magic method did better with the low energies, but less well with the higher energies compared to the non-magic-method sim. I don't know if that is expected. The sim was only run for 1 minute with 16 cores, so it probably isn't a good comparison.

MAGIC method low energy 
![low-ww](https://github.com/radiasoft/sirepo/assets/6516910/4c393fed-d698-455f-8267-b05c48a8707e)
MAGIC method higher energies
![high-ww](https://github.com/radiasoft/sirepo/assets/6516910/9cd62edd-a5b8-405e-941f-2ce73fab798e)

Normal sim low energy
![low-none](https://github.com/radiasoft/sirepo/assets/6516910/db47af49-7c75-4705-b258-69c2ecf23f1e)
Normal sim higher energies
![high-none](https://github.com/radiasoft/sirepo/assets/6516910/9a102fea-d21b-4a8f-9130-b0dacebe4368)
